influxql/iterator.go

```diff
@@ -618,6 +618,8 @@ func (a IteratorCreators) CreateIterator(opt IteratorOptions) (Iterator, error)
 			itr, err := ic.CreateIterator(opt)
 			if err != nil {
 				return err
+			} else if itr == nil {
```

Comment:
@jsternberg As an aside, I think it would be beneficial to document all these invariants around the query engine. Otherwise it's tricky for anyone else to work on the engine.